### ISSUE TITLE
async fn raises unused_mut on parameter that is used mutably

### ISSUE THREAD

**Issue by Arnavion** [rust-lang/rust#60566]
```
rustc 1.36.0-nightly (8dd4aae9a 2019-05-04)
binary: rustc
commit-hash: 8dd4aae9a83964cc08505da92d07ec68a3a2341d
commit-date: 2019-05-04
host: x86_64-pc-windows-msvc
release: 1.36.0-nightly
LLVM version: 8.0
```

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=4f2bb9f701bdbbb3424e926989bc0f2f

```rust
#![feature(async_await)]
#![deny(unused_mut)]

pub struct S;

pub async fn foo(mut s: S) {
    let _ = &mut s;
}
```

raises `unused_mut` on the `s` parameter of `foo`.

@taiki-e This is also raised in [the test case you added in #60501,](https://github.com/rust-lang/rust/pull/60501/files#diff-e7411cfc5acb4f1a88cde85379873c7d) but I assume the test passes because it defaults to ignoring the warning.

---

Workaround:

```diff
-pub async fn foo(mut s: S) {
+pub async fn foo(s: S) {
+    let mut s = s;
     let _ = &mut s;
```

**Comment by jonas-schievink** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/60498

**Comment by Arnavion** [comment]
@jonas-schievink Please read the issue carefully. Not only is it have a different issue from what #60498 was, I specifically linked to the PR that fixes #60498 with the intent of demonstrating that it's a different issue.

**Comment by jonas-schievink** [comment]
Oops, sorry!

**Comment by taiki-e** [comment]
> @taiki-e This is also raised in the test case you added in #60501, but I assume the test passes because it defaults to ignoring the warning.

Yes, I missed the warnings being ignored. This will be fixed in #60535. Also, there are reports in #60509 about other errors missed in #60501.
